### PR TITLE
hats example

### PR DESCRIPTION
**Pull request by amberroy**
Re-write of the Hats app using SDK 3.0.  Tracking skeleton not working atm so used the cursor ray (from cursor events) to get eye joint, this causes the hat to move slightly when the user moves their mouse.

### CODE REVIEW

File: examples/hats.html [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+				ray = event.ray;
+		  });
+		}
+
+		function start() {
+			updatePosition();
+			startAutoSend();
+		}
+
+		function startAutoSend() {
```
Comment (GavanWilhite) [inline examples/hats.html]: reason why not using the built in auto?

File: examples/hats.html [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+		for (var i=0; i < initialHats.length; i++) {
+			addEventListener(initialHats[i]);
+		}
+	}
+
+	function addEventListener(obj) {
+		obj.addEventListener('cursordown', function(event) {
+			if (selectedHat) sceneSync.destroy(selectedHat);
+			//destroyOnDisconnect=true so when user leaves, their hat vanishes.
+			var myHat = sceneSync.instantiate('Hat', {name: obj.name, sync: true}, true);
```
Comment (GavanWilhite) [inline examples/hats.html]: Cool this is so clean :-)

File: examples/hats.html (hunk 1 above)
Comment (amberroy) [inline examples/hats.html]: Ah yes should be using the builtin, I'll change it.  (this was a workaround for a bug that's now fixed: I'd forgotten to add the SyncScene behavior to the scene)

Comment (GavanWilhite) [inline examples/hats.html]: Actually, the built in may not work correctly might be best for now

File: examples/hats.html [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	};
+
+	var sim;
+	var sceneSync;
+	var objTemplateByName = {};
+	var selectedHat;
+	var initialHats = [];
+	var loadRequest;
+
+	var altPixelsPerMeter = 165.837418;//current Altspace default
```
Comment (GavanWilhite) [inline examples/hats.html]: Is there an issue getting this from altspace.getEnclosure() ?

**Comment by amberroy** [comment]
Closing, created a new branch.

**Pull request closed without merging**